core/src/main/java/com/zextras/carbonio/files/rest/services/PreviewService.java

```diff
@@ -235,23 +297,23 @@ public Try<BlobResponse> getThumbnailOfPdf(
    * @param requesterId is a {@link String} representing the id of the requester.
    * @param nodeId is a {@link String } representing the nodeId of the node.
    * @param version is a <code> integer </code> representing the version of the node.
-   * @param allowedMimeType is a {@link String} representing the allowed list or instance of
+   * @param supportedMimeTypeList is a {@link Set} representing the allowed list or instance of
    * mimetype that the calling methods allow (for instance a method may want only mimetype that are
    * of "image" so "image/something" while another method "application" so "application/something"
    *
    * @return a {@link Try} containing the checked @{link Node} or, on failure, the specific error
    */
-  private Try<Node> checkNodePermissionAndExistence(
+  public Try<Node> checkNodePermissionAndExistence(
```

Comment:
Since this method is used only by the `PreviewController`, I would move it there and I would make it private